sysdig/resource_sysdig_secure_custom_policy.go

```diff
@@ -123,14 +123,66 @@ func customPolicyToResourceData(policy *v2.Policy, d *schema.ResourceData) {
 		_ = d.Set("type", "falco")
 	}
 
-	rules := []map[string]interface{}{}
+	rules := getPolicyRulesFromResourceData(d)
+	newRules := []map[string]interface{}{}
 	for _, rule := range policy.Rules {
-		rules = append(rules, map[string]interface{}{
+		newRules = append(newRules, map[string]interface{}{
 			"name":    rule.Name,
 			"enabled": rule.Enabled,
 		})
 	}
-	_ = d.Set("rules", rules)
+	currentRules := []map[string]interface{}{}
+	for _, rule := range rules {
+		currentRules = append(currentRules, map[string]interface{}{
+			"name":    rule.Name,
+			"enabled": rule.Enabled,
+		})
+	}
+	fmt.Printf("Current rules: %+v", currentRules)
+	fmt.Printf("New rules: %+v", newRules)
+	areRulesSame := arePolicyRulesEquivalent(currentRules, newRules)
```

Comment:
Can we use `reflect.DeepEqual` provided by Go for checking equality of maps? (https://techwasti.com/the-best-way-to-check-the-equality-of-two-maps-is-in-go). It would be tested for all edge cases and reduce the code that we need to maintain

Reply:
I tried using that at first, took me forever to realize it wasn't working. The problem is the data structure is a slice of maps, and when doing reflect.DeepEquals it cares about the order of elements which is exactly what I am trying to avoid.